packages/queued-request-controller/src/QueuedRequestMiddleware.test.ts

```diff
@@ -1,607 +1,271 @@
-        middleware(
-          req,
-          {} as PendingJsonRpcResponse<typeof req>,
-          resolve,
-          reject,
-        ),
-      );
+    expect(mockEnqueueRequest).toHaveBeenCalledWith(
+      request,
+      expect.any(Function),
+    );
+  });
 
-      expect(mocks.enqueueRequest).toHaveBeenCalled();
-      // custom networks use getNetworkClientyId
-      expect(mocks.getNetworkClientById).toHaveBeenCalledWith(networkClientId);
+  it('calls next when a request is not queued', async () => {
+    const middleware = createQueuedRequestMiddleware({
+      enqueueRequest: getMockEnqueueRequest(),
+      useRequestQueue: () => false,
     });
+    const mockNext = jest.fn();
 
-    it('switchEthereumChain calls get queued but we dont check the current network', async () => {
-      const messenger = buildMessenger();
-      const middleware = createQueuedRequestMiddleware({
-        messenger,
-        useRequestQueue: () => true,
-      });
-      const mocks = buildMocks(messenger);
-
-      const req = {
-        ...requestDefaults,
-        method: 'wallet_switchEthereumChain',
-      };
-
-      await new Promise((resolve, reject) =>
-        middleware(
-          req,
-          {} as PendingJsonRpcResponse<typeof req>,
-          resolve,
-          reject,
-        ),
+    await new Promise((resolve) => {
+      mockNext.mockImplementation(resolve);
+      middleware(
+        getRequestDefaults(),
+        getPendingResponseDefault(),
+        mockNext,
+        jest.fn(),
       );
-
-      expect(mocks.addRequest).not.toHaveBeenCalled();
-      expect(mocks.enqueueRequest).toHaveBeenCalled();
-      expect(mocks.getProviderConfig).not.toHaveBeenCalled();
     });
 
-    describe('requiring switch', () => {
-      it('calls addRequest to switchEthChain if the current network is different than the globally selected network', async () => {
-        const messenger = buildMessenger();
-        const middleware = createQueuedRequestMiddleware({
-          messenger,
-          useRequestQueue: () => true,
-        });
-        const mockGetProviderConfig = jest.fn().mockReturnValue({
-          chainId: '0x5',
-        });
-        const mocks = buildMocks(messenger, {
-          getProviderConfig: mockGetProviderConfig,
-        });
-
-        const req = {
-          ...requestDefaults, // chainId = '0x1'
-          method: 'eth_sendTransaction',
-        };
-
-        await new Promise((resolve, reject) =>
-          middleware(
-            req,
-            {} as PendingJsonRpcResponse<typeof req>,
-            resolve,
-            reject,
-          ),
-        );
-
-        expect(mocks.addRequest).toHaveBeenCalled();
-        expect(mocks.enqueueRequest).toHaveBeenCalled();
-        expect(mocks.setNetworkClientIdForDomain).toHaveBeenCalled();
-      });
+    expect(mockNext).toHaveBeenCalled();
+  });
 
-      it('if the switchEthConfirmation is rejected, the original request is rejected', async () => {
```

Comment:
Do we want not still want to test this scenario?

Reply:
This scenario no longer has anything to do with the middleware. Network switching is part of `enqueueRequest`, so from the middleware's perspective, it appears the same as any other error thrown from `enqueueRequest`.

This is now tested in the controller tests, in the block `when the network switch for a single request fails`